std/traits.d

```diff
@@ -6490,6 +6509,38 @@ unittest
     }
 }
 
+/**
+Returns the number of full binary digits of precision representable by the scalar type `T`.
+ */
+template precision(T)
```

Comment:
A floating-point type `F` can losslessly represent all possible values of an integral type `N` if-and-only-if `precision!T >= precision!N`.

This trait is used by `smartOp` to help compute the optimal common numeric type for multi-argument functions like `binary()`.